aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -109,6 +109,35 @@ public void evaluate() throws Throwable {
         });
     }
 
+    @Test
+    public void timeIsConsumed() throws Exception {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "timeIsConsumed");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  timeout(time: 60, unit: 'SECONDS') {\n"
+                        + "    sleep 10\n"
+                        + "    semaphore 'timeIsConsumed'\n"
+                        + "    sleep 50\n"
```

Comment:
Maybe we could make these times shorter, so that test runs are quicker? IIUC running this test under heavy load will not cause it to fail; it would at worst cause it to _not_ fail if the bug fix were regressed. I suspect 10+10=20s would suffice to usually reproduce the bug.